### PR TITLE
View Switcher Improvements

### PR DESCRIPTION
**Pull request by prateekmedia**
- Use `AdwButton.flat` as base for View switcher tab
- Looks almost similar to the libadwaita one

View Switcher:
View Switcher

View Switcher Collapsed:
View Switcher collapased

Any feedback or code suggestion is welcome.

### CODE REVIEW

**Comment by jesusrp98** [comment]
I'm going to take a look at this PR now :)

**Comment by jesusrp98** [comment]
I can't install the latest libadwaita version that includes the updated view switcher. I'm on Fedora 35 and can't compile GTK 4.5 :(

Did you manage to test the new view switcher?

**Comment by prateekmedia** [comment]
@jesusrp98  Yes, You can try latest libadwaita flatpak if it's not compiling for you.

**Comment by jesusrp98** [comment]
> @jesusrp98 Yes, You can try latest libadwaita demo flatpak if it's not compiling for you.

Would you be able to provide me with a link? Thanks!

**Comment by jesusrp98** [comment]
I'm going to push a couple commits into this branch. We can change anything related to them no problem :)

File: lib/src/widgets/adw/view_switcher_tab.dart [hunk 1]
```diff
@@ -5,59 +5,73 @@ class AdwViewSwitcherTab extends StatelessWidget {
...
+      borderRadius: isDesktop
+          ? const BorderRadius.all(Radius.circular(6.0))
+          : BorderRadius.zero,
+      child: _AdwViewSwitcherTabLayout(
+        isRow: isDesktop,
+        children: [
+          if (data.icon != null)
+            Icon(
+              data.icon,
+              size: 18,
```
Comment (jesusrp98) [inline lib/src/widgets/adw/view_switcher_tab.dart:43]: I'm not 100% sure about this values. Did you check whether the default icon size for these widgets is `18`?

File: lib/src/widgets/adw/view_switcher_tab.dart [hunk 2]
```diff
@@ -5,59 +5,73 @@ class AdwViewSwitcherTab extends StatelessWidget {
...
             ),
+          if (data.icon != null && data.title != null)
+            const SizedBox(width: 8, height: 2),
```
Comment (jesusrp98) [inline lib/src/widgets/adw/view_switcher_tab.dart:46]: Same here, I think these values are correct, but not 100% sure.

**Review by jesusrp98 — COMMENTED** [review]
Apart from the couple comments I left, I think everything should be great now.

Let me know what you think about the latest commits I pushed, and whether you have any question or issues related to them.

**Comment by prateekmedia** [comment]
@jesusrp98 I only have one issue left for now and that is the code duplication part.

If someone wants to have ViewSwitcher for now then they have to create two different view switcher's: One for header bar one and the other for the Bottom Navigation Bar in `AdwScaffold` just like #46.

This type of code duplication is not there in the Original libadwaita so we need some way to unify both Flap and View Switcher so that the developer using this doesn't have to face the same.

File: lib/src/widgets/adw/view_switcher_tab.dart (hunk 1 above)
Comment (prateekmedia) [reply lib/src/widgets/adw/view_switcher_tab.dart:42]: They are a little smaller but I think it looks OK.

File: lib/src/widgets/adw/view_switcher_tab.dart (hunk 2 above)
Comment (prateekmedia) [reply lib/src/widgets/adw/view_switcher_tab.dart:45]: These are not 100% accurate but they look close to the libadwaita one.

**Comment by jesusrp98** [comment]
This issue is related, as you said, to the one related with `AdwSidebar`. Fixing that issue is out of scope of this PR, but we should definetly create an issue to detail how that could be fixed.

What do you think about it?

**Comment by prateekmedia** [comment]
@jesusrp98 Yes, That sounds good to me!

**Review by jesusrp98 — APPROVED** [review]

**Pull request merged**